### ISSUE TITLE
Thread Blocking fs.existsSync

### ISSUE THREAD

**Issue by robertmain** [robertmain/jukebox#116]
We are currently using `fs.existsSync` to check if a file exists on disk or not. This is not ideal as it's thread blocking.

### Platform
**OS:**  Windows
**Node Version:**  10.4.0
**NPM Version:**  6.1.0

#### Proposed solution
```js
const { promisify } = require('util');
const stat = promisify(fs.stat);
```

**Comment by robertmain** [comment]
Also look into the idea of just returning the stream and not handling the errors in there

**Comment by robertmain** [comment]
Closed in 5f1a538207b2f1668fab0982748658d1075a5776
